tests/unit/model-test.js

```diff
@@ -567,6 +569,106 @@ test('supports pushedData in root.deleted.uncommitted', function(assert) {
   });
 });
 
+test('supports canonical updates via pushedData in root.deleted.saved', function(assert) {
+  let { adapter } = env;
+
+  adapter.shouldBackgroundReloadRecord = () => false;
+  adapter.deleteRecord = () => {
+    return Ember.RSVP.resolve();
+  };
+
+  let record = run(() => store.push({
+    data: {
+      type: 'person',
+      id: '1',
+      attributes: {
+        isArchived: false
+      }
+    }
+  }));
+
+  run(() => {
+    record.destroyRecord().then(() => {
+      let currentState = record._internalModel.currentState;
+
+      assert.ok(currentState.stateName === 'root.deleted.saved',
+        'record is in a persisted deleted state');
+      assert.equal(get(record, 'isDeleted'), true);
+      assert.ok(store.peekRecord('person', '1') !== null, 'the deleted person is not removed from store (no unload called)');
+    });
+  });
+
+  run(() => {
+    try {
+      store.push({
+        data: {
+          type: 'person',
+          id: '1',
+          attributes: {
+            isArchived: true
+          }
+        }
+      });
+    } catch (e) {
+      assert.ok(false, e);
```

Comment:
what does this try/catch achieve?

Reply:
I think I can use `expectAssertion` here instead

Reply:
@hjdivad oh wait, actually, I can't unless there's a corollary? I basically want a "expect no assertion"